### PR TITLE
Move preferred_region boosting to function score (bug 1177718)

### PR DESCRIPTION
**Pull request by robhudson**

### CODE REVIEW

File: mkt/search/tests/test_views.py [hunk 1]
```diff
@@ -1346,6 +1346,18 @@ def test_search(self):
         eq_(objs[1]['title'], self.website.title)
         eq_(objs[1]['url'], self.website.url)
 
+    def test_search_preferred_region_match(self):
+        """
+        For websites, if the query string is something that will definitely not
+        match any websites we may still match on preferred_region. But we don't
+        want only preferred_region to find results.
+        """
+        self.website
```
Comment (diox) [inline mkt/search/tests/test_views.py]: remove this :)

Comment (robhudson) [inline mkt/search/tests/test_views.py]: hehe. oops. I was going to add a preferred region and then just used what was there. Good catch.

File: mkt/search/filters.py [hunk 2]
```diff
@@ -95,15 +95,20 @@ def filter_queryset(self, request, queryset, view):
...
-        if region:
-            should.append(query.Term(**{'preferred_regions': {
-                'value': region.id,
-                'boost': 4}}))
+        region_function = {
+            'filter': {'term': {'preferred_regions': region.id}},
+            # TODO: When we upgrade to Elasticsearch 1.4, change this to
+            # 'weight'
+            'boost_factor': 4,
+        }
```
Comment (robhudson) [inline mkt/search/filters.py]: There's no way I could find to do this using elasticsearch-dsl so I used the raw dict.

**Comment by diox** [comment]
r+

**Pull request merged**